connectors/sources/github.py

```diff
@@ -215,20 +369,297 @@ async def validate_config(self):
+                )
+            ):
+                if issue.get("pull_request") is not None:
+                    continue
+                document = self.prepare_document(
+                    github_document=issue, schema=ISSUE_SCHEMA
+                )
+                comments = []
+                async for comment in self.github_client.get_paginated_response(
+                    resource=ENDPOINTS["COMMENTS"].format(
+                        repo_name=repo_name, number=issue["number"], page_size=PAGE_SIZE
+                    )
+                ):
+                    comments.append(
+                        {"author": comment["user"]["login"], "body": comment["body"]}
+                    )
+                document.update(
+                    {
+                        "type": ObjectType.ISSUE.value,
+                        "owner": issue["user"]["login"],
+                        "repository": repo_name,
+                        "assignees": [
+                            assignee["login"] for assignee in issue["assignees"]
+                        ],
+                        "labels": [label["name"] for label in issue["labels"]],
+                        "comments": comments,
+                    }
+                )
+                yield document
+        except UnauthorizedException:
+            raise
+        except Exception as exception:
+            logger.warning(
+                f"Something went wrong while fetching the issues of {repo_name}. Exception: {exception}"
+            )
+
+    async def fetch_last_commit_timestamp(self, repo_name, path):
+        commits = await self.github_client.get_response(
+            resource=ENDPOINTS["COMMITS"].format(repo_name=repo_name, path=path)
+        )
+
+        # Get last commit timstamp from the list of the commits
+        return commits[0]["commit"]["committer"]["date"]  # pyright: ignore
+
+    async def fetch_files(self, repo_name, default_branch):
+        try:
+            file_tree = await self.github_client.get_response(
+                resource=ENDPOINTS["TREE"].format(
+                    repo_name=repo_name, default_branch=default_branch
+                )
+            )
+
+            for repo_object in file_tree.get("tree"):
+                last_commit_timestamp = await self.fetch_last_commit_timestamp(
+                    repo_name, repo_object["path"]
+                )
+                if repo_object["type"] == BLOB:
+                    file_name = repo_object["path"].split("/")[-1]
+                    repo_object.update(
+                        {
+                            "_timestamp": last_commit_timestamp,
+                            "repo_name": repo_name,
+                            "name": file_name,
+                            "extension": file_name[file_name.rfind(".") :]  # noqa
+                            if "." in file_name
+                            else "",
+                        }
+                    )
```

Comment:
```suggestion
                    file_name = repo_object["path"].split("/")[-1]
                    file_extension = file_name[file_name.rfind(".") :]  # noqa
                            if "." in file_name
                            else ""
                    repo_object.update(
                        {
                            "_timestamp": last_commit_timestamp,
                            "repo_name": repo_name,
                            "name": file_name,
                            "extension": file_extension,
                        }
                    )
```

nit: Would improve readability IMO.